models/user/user.go

```diff
@@ -120,6 +120,11 @@ type User struct {
 	// (ex: in private Gitea instances user won't be allowed to see even organizations/repositories that are set as public)
 	IsRestricted bool `xorm:"NOT NULL DEFAULT false"`
 
+	// Store the initial registration of the user, to aid in spam prevention
+	// Ensure that one IP isn't creating many accounts (following mediawiki approach)
+	InitialIP        string `xorm:"-"`
+	InitialUserAgent string `xorm:"-"`
```

Comment:
It shouldn't bloat the User model since they do not really belong to it.

Reply:
I agree it shouldn't bloat it, but my alternative was to add an additional input to the `CreateUser` function, and so I chose this route. I can adjust it if you prefer. Perhaps I can create a `UserMeta` struct that could be passed and reused for other things, too?

Reply:
I think UserMeta is right, but I won't strongly require it or block it.

Reply:
refactored to use a struct.